libs/MeshKernel/src/Mesh2D.cpp

```diff
@@ -1634,6 +1634,54 @@ std::vector<meshkernel::UInt> Mesh2D::GetHangingEdges() const
     return result;
 }
 
+std::vector<bool> Mesh2D::FilterBasedOnMetric(Location location,
+                                              Property metric,
+                                              double minValue,
+                                              double maxValue) const
+{
+    if (location != Location::Faces)
+    {
+        throw ConstraintError("Unsupported location. Only location faces is supported");
+    }
+    if (metric != Property::Orthogonality)
+    {
+        throw ConstraintError("Unsupported metric. Only orthogonality metric is supported");
+    }
+
+    const auto numFaces = GetNumFaces();
+
+    // Pre-allocate memory for result vector and set all elements to false
+    std::vector<bool> result(numFaces, false);
+
+    // Retrieve orthogonality values
+    const std::vector<double>& metricValues = GetOrthogonality();
+
```

Comment:
Re: the const reference here. Is this behaviour well defined?
The function GetOrthogonality does not return a reference.